### PR TITLE
docs: add apple privacy manifest code requirements

### PR DESCRIPTION
**Pull request by ItsChaceD**
This PR adds the [Apple Privacy Manifest Requirements](https://ionic.zendesk.com/hc/en-us/articles/22242418163735-Privacy-Manifest-Requirements) to the relevant plugins.

Addresses #2060

### CODE REVIEW

File: device/README.md [hunk 1]
```diff
@@ -9,7 +9,44 @@ npm install @capacitor/device
...
+
+### Example PrivacyInfo.xcprivacy
+
+```xml
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
+<plist version="1.0">
+  <dict>
+    <key>NSPrivacyTracking</key>
+    <false/>
```
Comment (OS-ricardomoreirasilva) [inline device/README.md:32]: Considering that this doesn't concern the required reason APIs, should we include it here?

Comment (OS-ricardomoreirasilva) [reply device/README.md:32]: This applies to all the `README` files on the PR.

File: device/README.md [hunk 2]
```diff
@@ -9,7 +9,44 @@ npm install @capacitor/device
...
+    <key>NSPrivacyAccessedAPITypes</key>
+    <array>
+      <!-- Add this dict entry to the array if the PrivacyInfo file already exists -->
+      <dict>
+        <key>NSPrivacyAccessedAPIType</key>
+        <string>NSPrivacyAccessedAPICategoryDiskSpace</string>
+        <key>NSPrivacyAccessedAPITypeReasons</key>
+        <array>
+          <string>85F4.1</string> <!-- Replace with the appropriate reason code -->
```
[the remark this answers is not in the document]
Comment (OS-ricardomoreirasilva) [reply device/README.md:41]: This applies to all the `README` files on the PR.

**Review by alexgerardojacinto — APPROVED** [review]

File: filesystem/README.md [hunk 3]
```diff
@@ -9,9 +9,42 @@ npm install @capacitor/filesystem
...
 
+## Apple Privacy Manifest Requirements
+
+Apple mandates that app developers now specify approved reasons for API usage to enhance user privacy. By May 1st, 2024, it's required to include these reasons when submitting apps to the App Store Connect.
+
+When using this specific plugin in your app, you must create a `PrivacyInfo.xcprivacy` file in `/ios/App` or use the VS Code Extension to generate it, specifying the usage reasons.
+
+For detailed steps on how to do this, please see the [Capacitor Docs](https://capacitorjs.com/docs/ios/privacy-manifest).
+
+**For this plugin, the required dictionary key is [NSPrivacyAccessedAPICategoryFileTimestamp](https://developer.apple.com/documentation/bundleresources/privacy_manifest_files/describing_use_of_required_reason_api#4278393) and the recommended reason is [CA617.1](https://developer.apple.com/documentation/bundleresources/privacy_manifest_files/describing_use_of_required_reason_api#4278393).**
```
Comment (jcesarmobile) [inline filesystem/README.md:20]:
```suggestion
**For this plugin, the required dictionary key is [NSPrivacyAccessedAPICategoryFileTimestamp](https://developer.apple.com/documentation/bundleresources/privacy_manifest_files/describing_use_of_required_reason_api#4278393) and the recommended reason is [C617.1](https://developer.apple.com/documentation/bundleresources/privacy_manifest_files/describing_use_of_required_reason_api#4278393).**
```

File: preferences/README.md [hunk 4]
```diff
@@ -23,7 +23,40 @@ npm install @capacitor/preferences
...
-## Example
+## Apple Privacy Manifest Requirements
+
+Apple mandates that app developers now specify approved reasons for API usage to enhance user privacy. By May 1st, 2024, it's required to include these reasons when submitting apps to the App Store Connect.
+
+When using this specific plugin in your app, you must create a `PrivacyInfo.xcprivacy` file in `/ios/App` or use the VS Code Extension to generate it, specifying the usage reasons.
+
+For detailed steps on how to do this, please see the [Capacitor Docs](https://capacitorjs.com/docs/ios/privacy-manifest).
+
+**For this plugin, the required dictionary key is [NSPrivacyAccessedAPICategoryUserDefaults](hhttps://developer.apple.com/documentation/bundleresources/privacy_manifest_files/describing_use_of_required_reason_api#4278401) and the recommended reason is [CA92.1](https://developer.apple.com/documentation/bundleresources/privacy_manifest_files/describing_use_of_required_reason_api#4278401).**
```
Comment (jcesarmobile) [inline preferences/README.md:34]:
```suggestion
**For this plugin, the required dictionary key is [NSPrivacyAccessedAPICategoryUserDefaults](https://developer.apple.com/documentation/bundleresources/privacy_manifest_files/describing_use_of_required_reason_api#4278401) and the recommended reason is [CA92.1](https://developer.apple.com/documentation/bundleresources/privacy_manifest_files/describing_use_of_required_reason_api#4278401).**
```

**Pull request merged**